src/pybind/mgr/dashboard/controllers/tcmu_iscsi.py

```diff
@@ -11,6 +11,14 @@
 @ApiController('tcmuiscsi')
 @AuthRequired()
 class TcmuIscsi(RESTController):
+    def _get_rate(self, daemon_type, daemon_name, stat):
+        data = mgr.get_counter(daemon_type, daemon_name, stat)[stat]
+
+        if data and len(data) > 1:
+            return (data[-1][1] - data[-2][1]) / float(data[-1][0] - data[-2][0])
+        else:
+            return 0
```

Comment:
`_get_rate` looks very similar to some existing functions:

* `mgr.dashboard.controllers.cephfs.CephFS#get_rate`
* `mgr.dashboard.controllers.osd.Osd#get_rate`
* `mgr.dashboard.controllers.perf_counters.PerfCounter#_get_rate`

Maybe, it would make sense to move `get_rate` to `mgr.dashboard.services.ceph_service.CephService` refactor the other occurrences?

Reply:
Please feel free to tackle that -- they used to all share the same function and during the refactor they all moved to duplicate implementations. I don't really have the bandwidth.

Reply:
@dillaman sure, that refactor should be done in a different PR by us. thanks